files/external_node_v2.rb

```diff
@@ -7,9 +7,14 @@
 # If --push-facts is given as the only arg, it uploads facts for all hosts and then exits.
 # Useful in scenarios where the ENC isn't used.
 
+require 'rbconfig'
 require 'yaml'
 
-$settings_file = "/etc/puppet/foreman.yaml"
+if RbConfig::CONFIG['host_os'] =~ /freebsd|dragonfly/i
+  $settings_file = "/usr/local/etc/puppet/foreman.yaml"
```

Comment:
I think we check this indirectly because we stub the loading of ```/etc/puppet/foreman.yaml```, so I think it should work on everything else we support. It does mean the tests should fail on freebsd/dragonfly.